bertopic/backend/_spacy.py

```diff
@@ -76,19 +76,23 @@ def embed(self,
         # Handle empty documents, spaCy models automatically map
         # empty strings to the zero vector
         empty_document = " "
+        embeddings = []
 
         # Extract embeddings from a transformer model
         if "transformer" in self.embedding_model.component_names:
-            embeddings = []
             for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
                 embeddings.append(self.embedding_model(doc or empty_document)._.trf_data.tensors[-1][0].tolist())
-            embeddings = np.array(embeddings)
 
         # Extract embeddings from a general spacy model
         else:
-            embeddings = []
             for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
                 embeddings.append(self.embedding_model(doc or empty_document).vector)
+
+        # See https://github.com/MaartenGr/BERTopic/issues/744 for context
+        # Convert to numpy arrays depending on whether cupy was used or not
+        if isinstance(embeddings, np.ndarray):
```

Comment:
I believe this will not work. `embeddings` will always be a list since it is initialized as a list and not changed during any of the code above. The commit that you referenced [here](https://github.com/MaartenGr/BERTopic/commit/a7927a2f7c3d18701ad275bdc232d00a21ca8baa#diff-06119c27943e751ff191ded5f03370df0e9e55afa3aeab96b8a2588ccb1cb6a0R97-R101) actually does it a bit different. It checks whether an element of `embeddings`, namely `embedding` is of the type `np.ndarray`. Also, note that the embeddings in the first `if` statement is converted to a list, so checking for a numpy array will not work.

Reply:
Ah, I didn't notice that, thank you